arxiv/metadata/metacheck.py

```diff
@@ -0,0 +1,687 @@
+        complaints.append("Authors: trailing whitespace")
+    #
+    if re.search( r"\s\s", v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: excess whitespace")
+    #
+    # This should also match ", ," ?
+    if re.search( r"\s,", v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: whitespace before comma")
+    #
+    if re.search( r"anonym", v, re.IGNORECASE):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: anonymous submissions not accepted")
+    #
+    if re.search( r"corresponding", v, re.IGNORECASE):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: contains 'corresponding'")
+    #
+    for s in ("\\dag", "\\ddag", "\\textdag", "\\textddag"):
+        if re.match( s, v, re.IGNORECASE):
+            disposition = combine_dispositions(disposition, WARN)
+            complaints.append("Authors: contains dagger symbol")
+            break
+        #
+    #
+    if re.match( "authors", v, re.IGNORECASE):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: begins with 'authors'")
+    elif re.match( "author", v, re.IGNORECASE):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: begins with 'author'")
+    #
+    if contains_html_elements(v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: contains HTML")
+    #
+    if not all_brackets_balanced(v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: unbalanced brackets")
+    #
+    # Do I need to parse the authors list first?
+    if re.search( r";", v) and not re.search( "(.*;.*)", v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: use commas, not ';' to separate authors")
+    #
+    # Authors list can NOT end with punctuation
+    if ends_with_punctuation(v):
+        disposition = combine_dispositions(disposition, WARN)
+        complaints.append("Authors: ends with punctuation")
+    #
+    
+    # Parse authors, check authors names - but not affiliations - for:
+    # ALL CAPS
+    # single word in all CAPS (not really done)
+    # [...] (in author name)
+    # Numbers (but Jennifer 8 Lee!)
+    # IEEE (confusable with affiliation)
+    # phd, prof, dr
+    # Single initial after parsing authors?
+    # TODO: letters after name (BUT Indian names like RS, etc?)
+    # TODO: extra curlies in TeX accents, e.g.: {\'e}
+    # (TODO: allow all Unicode ?)
+    # TODO: urls
+
+    # TODO: maybe remove duplicates from the complaints list
+    
+    parsed_authors = parse_author_affil(v) # => List[List[str]]
+    for author in parsed_authors:
+        # first is "keyname" second is fist name(s), third is suffix

```

Comment:
Consider splitting out this author name check to its own function.

Reply:
Done.